fdbserver/ClusterController.actor.cpp

```diff
@@ -1691,6 +1704,216 @@ class ClusterControllerData {
 		return idUsed;
 	}
 
+    // Updates work health signals in `workerHealth` based on `req`.
+	void updateWorkerHealth(const NetworkAddress& workerAddress, const std::vector<NetworkAddress>& degradedPeers) {
+		std::string degradedPeersString;
+		for (int i = 0; i < degradedPeers.size(); ++i) {
+			degradedPeersString += i == 0 ? "" : " " + degradedPeers[i].toString();
+		}
+		TraceEvent("ClusterControllerUpdateWorkerHealth")
+		    .detail("WorkerAddress", workerAddress)
+		    .detail("DegradedPeers", degradedPeersString);
+
+		// `req.degradedPeers` contains the latest peer performance view from the worker. Clear the worker if the
+        // requested worker doesn't see any degraded peers.
+		if (degradedPeers.empty()) {
+			workerHealth.erase(workerAddress);
+			return;
+		}
+
+		double currentTime = now();
+
+        // Current `workerHealth` doesn't have any information about the incoming worker. Add the worker into
+        // `workerHealth`.
+		if (workerHealth.find(workerAddress) == workerHealth.end()) {
+			workerHealth[workerAddress] = {};
+			for (const auto& degradedPeer : degradedPeers) {
+				workerHealth[workerAddress].degradedPeers[degradedPeer] = { currentTime, currentTime };
+			}
+
+			return;
+		}
+
+		// The incoming worker already exists in `workerHealth`.
+
+		auto& health = workerHealth[workerAddress];
+
+		// First, remove any degraded peers recorded in the `workerHealth`, but aren't in the incoming request. These
```

Comment:
the format looks funny. 
likely tab vs space. 
can you run clang-format to the change?